server/src/test/java/org/opensearch/cluster/coordination/JoinTaskExecutorTests.java

```diff
@@ -327,6 +362,399 @@ public void testJoinClusterWithDecommissionFailed() {
         JoinTaskExecutor.ensureNodeCommissioned(discoveryNode, metadata);
     }
 
+    public void testJoinClusterWithNonRemoteStoreNodeJoining() {
+        DiscoveryNode joiningNode = newDiscoveryNode(Collections.emptyMap());
+        ClusterState currentState = ClusterState.builder(ClusterName.DEFAULT)
+            .nodes(DiscoveryNodes.builder().add(joiningNode).build())
+            .build();
+
+        JoinTaskExecutor.ensureNodesCompatibility(joiningNode, currentState.getNodes(), currentState.metadata());
+    }
+
+    public void testJoinClusterWithRemoteStoreNodeJoining() {
```

Comment:
In all these tests, joiningNode and nodes in currentState is same node itself. Is this intended?